src/views/tutorial-view/components/image-config/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { Children, isValidElement, ReactElement, ReactNode } from 'react'
+import MdxImg from '../mdx-img'
+// import classNames from 'classnames'
+import s from './image-config.module.css'
+
+/**
+ * Configure extra settings for images via MDX.
+ * Designed to wrap a single markdown image.
+ */
+export default function ImageConfig({
+  /** Hide borders (temporarily configurable while we remove baked-in borders from existing images) */
+  hideBorder = false,
+  /** A caption to display below the image. */
+  caption,
+  /** An MDX `img`, e.g. `![some alt text](path/to/img.jpg)` */
+  children,
+}: {
+  caption?: string
+  hideBorder?: boolean
+  children: ReactNode
+}): ReactElement {
+  //  emit a warning if no meaningful props are present
+  if (!hideBorder && !caption) {
+    console.warn(
+      `Warning: <ImageConfig /> was initialized with default props. Please remove <ImageConfig /> if configuration is not needed.`
+    )
+  }
+
+  // TODO: fully deprecate the `hideBorder` prop, then remove this warning
```

Comment:
I feel like this TODO has been here for a while. Is this deprecation something we can feasibly do? And if so, is it being actively worked on? My assumption is the answer to both is "maybe" rather than a clear "yes", so for now I've retained the `hideBorder` functionality (despite being very on board with deprecating it).

Reply:
I don't have context on the status of deprecation, but it does seem feasible to do IMO. Just did a search on Sourcegraph and it looks like `hideBorder` is still commonly used.

Reply:
Oh! just saw this note, I added another one in the 'files' view. This deprecation happened before my time. Actually deprecating it would require us to collaborate with content authors to remove the instances of it in the `tutorials` repo.

Reply:
After all the convo we've had on this so far, honestly it seems like we can just totally remove the prop in DevDot and not add a deprecation warning, as long as we have an Asana task for removing all uses of it in the tutorials repo. This prop seems like an easy one to locate usage of with VS Code's search or Sourcegraph's.

Reply:
☝️ @ashleemboyer Base on your comment on the docs situation, I think we should hold off on removing `hideBorder` until we handle the `docs` use case. I believe in `docs` we'll need it, and we may even want to remove the deprecation warning 🤔 . I think we could conditionally render a warning for Learn content, or even ignore the `hideBorder` prop but only in Learn contexts, if the education team is taking a harder line on never using the prop.